test/hw9/test_task3.py

```diff
@@ -0,0 +1,58 @@
+import tempfile
+
+import pytest
+
+from All_home_works.hw9.task3 import universal_file_counter
+
+
+@pytest.fixture()
+def make_dir():
+    directory = None
+    files = []
+
+    def _dirname(data):
+        nonlocal directory, files
+        directory = tempfile.TemporaryDirectory()
+        for extension in (".py", ".txt", ".txt"):
+            named_file = tempfile.NamedTemporaryFile(
+                dir=directory.name, suffix=extension, mode="w"
+            )
+            named_file.writelines(data)
+            named_file.seek(0)
+            files.append(named_file)
+        return directory.name
+
+    yield _dirname
+    for file in files:
+        file.close()
```

Comment:
Can we close files using context manager?

Reply:
If I remove these lines it will be an error: ` No such file or directory: '/tmp/...`.
I think it Is because in the actual function we are opening this file and then closing it. And while we are closing this file, the file deletes itself. And when the directory is cleaning up it is again trying to delete this not existing file, so I am catching the error.